src/components/ImageView/index.native.js

```diff
@@ -66,12 +66,18 @@ class ImageView extends PureComponent {
         if (!this.props.url) {
             return;
         }
-        ImageSize.getSize(this.props.url).then(({width, height}) => {
+        ImageSize.getSize(this.props.url).then(({width, height, rotation}) => {
             let imageWidth = width;
             let imageHeight = height;
             const containerWidth = Math.round(this.props.windowWidth);
             const containerHeight = Math.round(this.state.containerHeight);
 
+            // On specific Android devices, the dimensions are sometimes returned to us flipped here, with `rotation` set to 90 degrees.
+            // Swap them back to make sure the image fits nicely in the container. On iOS, the rotation is always undefined, so this does not apply.
+            if ((rotation === 90 || rotation === 270) && imageWidth > imageHeight) {
```

Comment:
I'm thinking that this `&& imageWidth > imageHeight` is probably not needed. I think a regular camera usually takes 4:3 (width:height) images, and  store rotation information. This condition would make it miss your `if` when it really should enter.

This is a less common case, but I guess we should still handle it correctly if possible. I will try to find such picture and upload it next week.

Reply:
Try with this image:

https://github.com/recurser/exif-orientation-examples/blob/master/Landscape_8.jpg?raw=true

Removing `&& imageWidth > imageHeight` results in the image using the maximum space